Scala/Hangman.scala

```diff
@@ -0,0 +1,294 @@
+              outString += theWord.toCharArray.charAt(t).toString
+            }
+            outString
+          } + "', stupid.)")
+
+
+          println("\nTry again? Y/N\n")
+          var newGame: String = scala.io.StdIn.readLine().toLowerCase()
+          newGame match
+          {
+            case "y" =>
+            case _ =>
+              println("\nGoodbye. Try again some other time!\n")
+              gameSwitch = 1
+          }
+          endSwitch = 1
+
+        case _ =>
+
+          var hideWord = theWord.toBuffer
+
+          for (z <- 0 to hideWord.length - 1) {
+            hideWord(z) = '_'
+          }
+
+          var outWord: String = ""
+
+          for (y <- 0 to theWord.length - 1) {
+            for (x <- 0 to usedLetters.length - 1) {
+              if (usedLetters.charAt(x) == theWord.charAt(y)) {
+                hideWord(y) = usedLetters.charAt(x)
+              }
+
+            }
+          }
+
+          var checkWord = ""
+          for (u <- 0 to hideWord.length - 2) {
+            checkWord += hideWord(u).toString
+          }
+
+          checkWord matches ".*_.*" match {
+
+            case false =>
+
+              println("")
+              println("------")
+              println("|  |")
+              println("| (_)")
+              println("|       \\_O_/")
+              println("|         |    ")
+              println("|_____  _/ \\_")
+
+              println("\nYou win! Success!")
+
+
+
+              println("\nTry again? Y/N\n")
+              var newGame: String = scala.io.StdIn.readLine().toLowerCase()
+              newGame match
+              {
+                case "y" =>
+                case _ =>
+                  println("\nGoodbye. Try again some other time!\n")
+                  gameSwitch = 1
+              }
+
+              endSwitch = 1
+
+
+            case true =>
+
+              (6 - hangCount) match {
+                case 1 =>
+                  println("\n" + (6 - hangCount + " mistake and you hang!\n\n" + {
+                    for (w <- 0 to hideWord.length - 2) {
+                      outWord += hideWord(w).toString + " "
+                    };
+                    outWord
+                  }))
+
+                case _ =>
+                  println("\n" + (6 - hangCount + " mistakes and you hang!\n\n" + {
+                    for (w <- 0 to hideWord.length - 2) {
+                      outWord += hideWord(w).toString + " "
+                    };
+                    outWord
+                  }))
+              }
+
+
+              println("\nYou have used the following letters: " + usedLetters)
+
+              println("\nPlease choose a letter.\n")
+              var input: String = scala.io.StdIn.readLine()
+
+              usedLetters += input
+
+              theWord.toCharArray.contains(input.charAt(0)) match {
+                case false => hangCount += 1
+                case true =>
+              }
+
+              hangCount match {

```

Comment:
These sort of static drawings could be a separate method outside the current one